src/sas/sasgui/perspectives/fitting/basepage.py

```diff
@@ -1246,12 +1244,11 @@ def reset_page_helper(self, state):
 
         # we have two more options for smearing
         if self.pinhole_smearer.GetValue():
-            self.dx_min = state.dx_min
-            self.dx_max = state.dx_max
-            if self.dx_min is not None:
-                self.smear_pinhole_min.SetValue(str(self.dx_min))
-            if self.dx_max is not None:
-                self.smear_pinhole_max.SetValue(str(self.dx_max))
+            self.dx_percent = state.dx_percent
+            if self.dx_percent is not None:
+                if state.dx_old:
+                    self.dx_percent = self.data.x[0] / self.dx_percent
```

Comment:
Should probably clear dx_old since dx_percent no longer contains the dx_min from the saved state, and now contains a ratio.

Also, shouldn't this be 100*ratio rather than ratio since you divide by 100 before applying it?